lib/zenaton/client.rb

```diff
@@ -64,23 +64,24 @@ def initialize
 
     # Gets the url for the workers
     # @param resource [String] the endpoint for the worker
-    # @param params [String] url encoded parameters to include in request
+    # @param params [Hash] query params to be url encoded
     # @return [String] the workers url with parameters
-    def worker_url(resource = '', params = '')
+    def worker_url(resource = '', params = {})
```

Comment:
The agent library use this method, and the one bellow. And it sends parameters as a string.
If you take a look at [the PHP implementation](https://github.com/zenaton/zenaton-php/pull/22), I kept the default value as an empty string, and have two different behaviors, depending on whether the given parameter is a string or a map.
I think you will need to do something similar to make sure the agent will not break if users update their library without updating their agent.

Reply:
I was hoping I would be able to only update the agent to act differently based on the gem version. But I forgot about the scenario you mentioned ... :disappointed:

Reply:
OK I tried a slightly different approach by keeping the old behaviour but adding a warning. The warning would then only be printed when the gem has been updated but not the agent. But it should still work as it did before.